assets/scripts/build_gallery.py

```diff
@@ -365,113 +496,59 @@ def build_gallery(
                 print(traceback.format_exc())
                 expectation_tracebacks.write(
-                    f"\n\n----------------\n[JSON write fail] {expectation} ({group})\n"
+                    f"\n\n----------------\n[JSON write fail] {expectation} ({expectations_info[expectation]['package']})\n"
                 )
                 expectation_tracebacks.write(traceback.format_exc())
 
-    if just_installed:
-        print("\n\n\n=== (Uninstalling) ===")
-        logger.info(
-            f"Uninstalling packages that were installed while running this script..."
-        )
-        for req in just_installed:
-            logger.debug(f"Executing command: 'pip uninstall -y \"{req}\"'")
-            execute_shell_command(f'pip uninstall -y "{req}"')
-
-    expectation_filenames_set = set(requirements_dict.keys())
-    full_registered_expectations_set = set(
-        great_expectations.expectations.registry.list_registered_expectation_implementations()
-    )
-    if only_these_expectations:
-        registered_expectations_set = (
-            set(only_these_expectations) & full_registered_expectations_set
-        )
-        expectation_filenames_set = (
-            set(only_these_expectations) & expectation_filenames_set
-        )
-    elif not include_core:
-        registered_expectations_set = full_registered_expectations_set - set(
-            core_expectations
-        )
-    else:
-        registered_expectations_set = full_registered_expectations_set
-    non_matched_filenames = (
-        expectation_filenames_set - registered_expectations_set - failed_to_import_set
-    )
-    if failed_to_import_set:
-        expectation_tracebacks.write(f"\n\n----------------\n(Not a traceback)\n")
-        expectation_tracebacks.write("Expectations that failed to import:\n")
-        for expectation in sorted(failed_to_import_set):
-            expectation_tracebacks.write(f"- {expectation}\n")
-    if non_matched_filenames:
-        expectation_tracebacks.write(f"\n\n----------------\n(Not a traceback)\n")
-        expectation_tracebacks.write(
-            "Expectation filenames that don't match their defined Expectation name:\n"
-        )
-        for fname in sorted(non_matched_filenames):
-            expectation_tracebacks.write(f"- {fname}\n")
+    # Iterate the gallery_info_by_backend dict and write the backend-specific files
+    for _backend in gallery_info_by_backend:
+        if gallery_info_by_backend[_backend]:
+            with open(f"{_backend}_{backend_outfile_suffix}.json", "w") as outfile:
+                json.dump(gallery_info_by_backend[_backend], outfile, indent=4)
 
-        bad_names = sorted(
-            list(registered_expectations_set - expectation_filenames_set)
-        )
-        expectation_tracebacks.write(
-            f"\nRegistered Expectation names that don't match:\n"
+    # Only attempt to combine and write to file when no Expectations are skipped
+    if backend_outfile_suffix == "full":
```

Comment:
Just trying to understand - is this not checking the `only_combine` flag because we are also combining on each run of build_gallery()?